testing/xfpga/test_feature_open_c.cpp

```diff
@@ -205,14 +211,22 @@ TEST_P(feature_open_c_p, test_feature_functions) {
 	fpga_feature_properties feature_prop;
 	EXPECT_EQ(xfpga_fpgaFeaturePropertiesGet(ftokens_[0], &feature_prop), FPGA_OK);
 
+	wrapped_feature_token *wt = (wrapped_feature_token *)ftokens_[0];
+	fpga_feature_token tmp = wt->feature_token;
+	((struct _fpga_feature_token *)tmp)->magic = FPGA_FEATURE_TOKEN_MAGIC + 1;
+	EXPECT_EQ(xfpga_fpgaFeaturePropertiesGet(ftokens_[0], &feature_prop), FPGA_INVALID_PARAM);
+
+	((struct _fpga_feature_token *)tmp)->magic = FPGA_FEATURE_TOKEN_MAGIC;
 	EXPECT_EQ(xfpga_fpgaFeatureOpen(ftokens_[0], 0, nullptr, &feature_h), FPGA_OK);
-	
+
 	fpga_dma_properties dma_prop;
 	EXPECT_EQ(xfpga_fpgaDMAPropertiesGet(ftokens_[0], &dma_prop), FPGA_OK);
-	
+
 	dma_transfer_list t_list;
 	EXPECT_EQ(xfpga_fpgaDMATransferSync(feature_h, &t_list), FPGA_OK);
-	
+
+	uint64_t context = 0;
+	EXPECT_EQ(xfpga_fpgaDMATransferAsync(feature_h, &t_list, dma_cb, (void*)&context), FPGA_OK);
```

Comment:
We should verify that the callback fn was actually called and that it was called with the correct context.

Reply:
Currently the DMA plugin layer is not implemented and I just put a stub DMA plugin module for placement holder. The verification of the callback function will be checked when its definition is clear.